### PR TITLE
[Caffe2] Changes done inside Facebook

### PR DESCRIPTION
**Pull request by yinghai**
Sync fbcode -> github

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -87,6 +87,10 @@
 #     we will search for libraries in these paths
 
 
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
```
Comment (yinghai) [inline setup.py]: @ezyang Need an thumb-up on this.

**Review by orionr — APPROVED** [review]
Looks good! Thanks.

**Comment by orionr** [comment]
@soumith could you approve the setup.py changes here instead? @ezyang shouldn't need to do it right now. :) Thanks.

**Comment by ezyang** [comment]
Are the `__future__` insertions due to a lint rule in fbcode? They seem harmless but the prevailing PyTorch style seems to not do future imports unless they actually would matter.

**Comment by orionr** [comment]
Yup - those were added as a part of a fbcode lint rule. We can back them out if we want.

**Review by ezyang — APPROVED** [review]

**Comment by yinghai** [comment]
@orionr @ezyang For the record, adding those 4 lines of import in `setup.py` will result in build failures. 
```
(venv)[yinghai:pytorch: (fbsync)]$ python setup.py install
Traceback (most recent call last):
  File "setup.py", line 800, in <module>
    extra_link_args=extra_link_args + main_link_args + [make_relative_rpath('lib')],
  File "/data/users/yinghai/aml/venv/lib/python2.7/site-packages/setuptools/extension.py", line 39, in __init__
    _Extension.__init__(self, name, sources, *args, **kw)
  File "/usr/lib64/python2.7/distutils/extension.py", line 106, in __init__
    assert type(name) is StringType, "'name' must be a string"
AssertionError: 'name' must be a string
```

**Comment by yinghai** [comment]
onnx-fb-universe failure is due to missing image.

**Pull request merged**